### PR TITLE
Add docker image, add auto deploy

### LINKED ISSUE #23: Automatic deployer via docker image
Want to be able to spin up and deploy a contract to a local ganache and get the address of the deployed contract.

### PR DESCRIPTION
**Pull request by gaieges**
Closes #23

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    environment:
+      - ETH_HOST=ganache
+      - ETH_PORT=8545
+      - ADDRESS_OUTPUT_FILENAME=/deployed/ethkvstore.address.json
+    depends_on:
+      - ganache
+    volumes:
+      - deployed:/deployed
+
+  # I'll get rid of this, just wanna see it work
```
Comment (rbval) [inline docker-compose.yml]: As you mentioned in the comments, let's make sure this one is removed before merging.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    build: .
+    image: hcaptcha/eth-kvstore
+    environment:
+      - ETH_HOST=ganache
+      - ETH_PORT=8545
+      - ADDRESS_OUTPUT_FILENAME=/deployed/ethkvstore.address.json
+    depends_on:
+      - ganache
+    volumes:
+      - deployed:/deployed
```
Comment (rbval) [inline docker-compose.yml]: We don't need to specify the volume in here.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  # I'll get rid of this, just wanna see it work
+  test:
+    image: alpine
+    command: sh -c " while [ ! -f /deployed/ethkvstore.address.json ]; do sleep 1 ;done ; cat /deployed/ethkvstore.address.json"
+    depends_on:
+      - kvstore
+    volumes:
+      - deployed:/deployed
+
+volumes:
```
Comment (rbval) [inline docker-compose.yml]: This is not needed either.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    ports:
+      - 8545:8545
+
+  kvstore:
+    build: .
+    image: hcaptcha/eth-kvstore
+    environment:
+      - ETH_HOST=ganache
+      - ETH_PORT=8545
+      - ADDRESS_OUTPUT_FILENAME=/deployed/ethkvstore.address.json
```
Comment (rbval) [inline docker-compose.yml]: This is not needed here either.

**Comment by gaieges** [comment]
in order for those containers to share the file, we do need a mount of sorts, that’s what the other repo is doing. how would you do that differently? 

but you make me realize i should add a test to verify it gets output correctly vs testing it via docker compose

**Comment by rbval** [comment]
I would do it the same way as in hmt-token or change hmt-token to be similar to this one.

**Comment by rbval** [comment]
I understand the mounting, but in hmt-token it’s not done in its own docker-compose.

**Comment by gaieges** [comment]
Should be all fixed up now robert

**Pull request merged**